app/src/main/java/com/cornellappdev/coffee_chats_android/CreateProfileFragment.kt

```diff
@@ -115,79 +114,62 @@ class CreateProfileActivity : AppCompatActivity() {
-            // update information on backend
-            val pronouns = pronounSpinner.selectedItem as String
-            val graduationYear = (classSpinner.selectedItemPosition + year).toString()
-            val major = majorACTV.text.toString()
-            val hometown = hometownEditText.text.toString()
-            val demographics = Demographics(pronouns, graduationYear, major, hometown, "")
-
-            val updateDemographicsEndpoint = Endpoint.updateDemographics(demographics)
-            val typeToken = object : TypeToken<ApiResponse<Demographics>>() {}.type
-            CoroutineScope(Dispatchers.IO).launch {
-                val updateDemographicsResponse = Request.makeRequest<ApiResponse<Demographics>>(
-                    updateDemographicsEndpoint.okHttpRequest(),
-                    typeToken
-                )
-                if (updateDemographicsResponse == null || !updateDemographicsResponse.success) {
-                    Toast.makeText(applicationContext, "Failed to save information", Toast.LENGTH_LONG).show()
-                }
-            }
-
-            val intent = Intent(this, GroupInterestActivity::class.java)
-            intent.putExtra("page", 1)
-            startActivity(intent)
-            overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_left)
-        }
     }
 
     // checks whether the major and hometown fields are completed, enabling or disabling the next button accordingly
     private fun toggleNextButton() {
         if (majorFilled && hometownFilled) {
-            nextButton.isEnabled = true
-            nextButton.isClickable = true
+            callback!!.onFilledOut()
         } else {
-            nextButton.isEnabled = false
-            nextButton.isClickable = false
+            callback!!.onSelectionEmpty()
+        }
+    }
+
+    private var callback: OnFilledOutListener? = null
+
+    override fun setOnFilledOutListener(callback: OnFilledOutListener) {
+        this.callback = callback
+    }
+
+    override fun saveInformation() {
+        val pronouns = pronounSpinner.selectedItem as String
+        val graduationYear = (classSpinner.selectedItemPosition + year).toString()
+        val major = majorACTV.text.toString()
+        val hometown = hometownEditText.text.toString()
+        val demographics = Demographics(pronouns, graduationYear, major, hometown, "")
+
+        val updateDemographicsEndpoint = Endpoint.updateDemographics(demographics)
+        val typeToken = object : TypeToken<ApiResponse<Demographics>>() {}.type
+        CoroutineScope(Dispatchers.IO).launch {
+            val updateDemographicsResponse = Request.makeRequest<ApiResponse<Demographics>>(
+                updateDemographicsEndpoint.okHttpRequest(),
+                typeToken
+            )
+            if (updateDemographicsResponse == null || !updateDemographicsResponse.success) {
+                Toast.makeText(requireContext(), "Failed to save information", Toast.LENGTH_LONG)
```

Comment:
Eventually, consider moving away from Toast as our primary source of popup information. Snackbar or a custom solution could look better, depending on the situation or priority.